File: DavidNyamutale_miniproject_deliverable_Number_four/Python Module that imports Audio Library/AudioRecorderPlayerModule.py
import jpype
#from JVMWrapper import JVMWrapper

# Start the JVM with the path to AudioLibrary.jar file
jpype.startJVM(classpath=[r"C:\Users\David Claudian\Documents\NetBeansProjects\AudioLibrary\dist\AudioLibrary.jar"])

# Import Java classes
AudioLibrary = jpype.JClass("audiolibrary.AudioLibrary")

# Connect to the MySQL database
connection = jpype.JClass("java.sql.DriverManager").getConnection("jdbc:mysql://localhost:3306/audio_library?zeroDateTimeBehavior=convertToNull", "root", "")

try:
    while True:
        print("\n1. Record audio")
        print("2. Play recorded audio")
        print("3. Exit")
        choice = input("Enter your choice (1/2/3): ").strip()

        if choice == "1":
            name = input("Enter a name for the recording: ").strip()
            duration = int(input("Enter the duration of the recording in seconds: ").strip())
            audio_data = AudioLibrary.recordAudio(duration)
            AudioLibrary.saveAudioToDatabase(connection, name, audio_data)
        elif choice == "2":
            play_name = input("Enter the name of the recording to play: ").strip()
            AudioLibrary.playAudioFromDatabase(connection, play_name)
        elif choice == "3":
            connection.close()
            jpype.shutdownJVM()
            exit(0)
        else:
            print("Invalid choice. Please enter 1, 2, or 3.")

except jpype.JException as ex:
    ex.printStackTrace()

except KeyboardInterrupt:
    # Handle keyboard interrupt (Ctrl+C) to ensure JVM shutdown
    connection.close()
    jpype.shutdownJVM()

except Exception as e:
    print("An error occurred:", str(e))

finally:
    jpype.shutdownJVM()

File: DavidNyamutale_miniproject_deliverable_Number_four/Java module that imports Audio Library/AudioRecorderPlayerModule.java
package audiorecorderplayermodule;

import audiolibrary.AudioLibrary;
import java.io.IOException;

import java.sql.Connection;
import java.sql.DriverManager;
import java.sql.SQLException;
import java.util.Scanner;
import javax.sound.sampled.LineUnavailableException;
import javax.sound.sampled.UnsupportedAudioFileException;

/**
 * A simple project demonstrating the use of AudioLibrary for audio recording and playback.
 */
public class AudioRecorderPlayerModule {

    /**
     * Main method to run the audio recording and playback application.
     *
     * @param args Command-line arguments (not used).
     */
    public static void main(String[] args) throws LineUnavailableException, IOException, UnsupportedAudioFileException {
        Connection connection = null;

        try {
            // Connect to MySQL database
            connection = DriverManager.getConnection("jdbc:mysql://localhost:3306/audio_library?zeroDateTimeBehavior=convertToNull", "root", "");
            Scanner scanner = new Scanner(System.in);

            while (true) {
                System.out.println("\n1. Record audio");
                System.out.println("2. Play recorded audio");
                System.out.println("3. Exit");
                System.out.print("Enter your choice (1/2/3): ");

                if (scanner.hasNextLine()) {
                    String choice = scanner.nextLine().trim();

                    switch (choice) {
                        case "1":
                            System.out.print("Enter a name for the recording: ");
                            String name = scanner.nextLine().trim();
                            System.out.print("Enter the duration of the recording in seconds: ");
                            int duration = Integer.parseInt(scanner.nextLine().trim());
                            byte[] audioData = AudioLibrary.recordAudio(duration);
                            AudioLibrary.saveAudioToDatabase(connection, name, audioData);
                            break;
                        case "2":
                            System.out.print("Enter the name of the recording to play: ");
                            String playName = scanner.nextLine().trim();
                            AudioLibrary.playAudioFromDatabase(connection, playName);
                            break;
                        case "3":
                            connection.close();
                            System.exit(0);
                            break;
                        default:
                            System.out.println("Invalid choice. Please enter 1, 2, or 3.");
                    }
                } else {
                    System.out.println("No input found. Please try again.");
                }
            }

        } catch (SQLException e) {
            e.printStackTrace();
        }
    }
}
